mitmproxy/net/http/request.py

```diff
@@ -468,7 +468,8 @@ def _get_multipart_form(self):
         return ()
 
     def _set_multipart_form(self, value):
-        raise NotImplementedError()
+        self.headers["content-type"] = "multipart/form-data"
+        self.content = mitmproxy.net.http.url.encode(value, self.get_text(strict=False)).encode()
```

Comment:
This sets a multipart header but an urlencode value. They are not the same, so this won't fly! This is akin to taking a Word document and changing the filename to ".pdf". It won't make it a PDF document either!